Pythonwin/pywin/framework/scriptutils.py

```diff
@@ -602,20 +603,19 @@ def JumpToDocument(fileName, lineno=0, col=1, nChars=0, bScrollToTop=0):
     return view
 
 
-def _HandlePythonFailure(what, syntaxErrorPathName=None):
+def _HandlePythonFailure(what: str, syntaxErrorPathName: str | None = None):
     typ, details, tb = sys.exc_info()
     if isinstance(details, SyntaxError):
+        filename = details.filename
+        if (not filename or filename == "<string>") and syntaxErrorPathName:
+            filename = syntaxErrorPathName
         try:
-            msg, (fileName, line, col, text) = details
```

Comment:
This code would produce:
`TypeError: cannot unpack non-iterable SyntaxError object`